cpp/src/arrow/filesystem/s3_internal.h

```diff
@@ -291,6 +294,42 @@ class ConnectRetryStrategy : public Aws::Client::RetryStrategy {
   int32_t max_retry_duration_;
 };
 
+/// \brief calculate the MD5 of the input sse-c key(raw key, not base64 encoded)
+/// \param sse_customer_key is the input sse key
+/// \return the base64 encoded MD5 for the input key
+inline Result<std::string> CalculateSSECKeyMD5(const std::string& sse_customer_key) {
```

Comment:
Could you avoid abbreviating here like others?

```suggestion
inline Result<std::string> CalculateSSECustomerKeyMD5(const std::string& sse_customer_key) {
```